### ISSUE TITLE
Document how block and inline levels work

### ISSUE THREAD

**Issue by ndtodoroff** [typst/typst#409]
Perhaps this is a documentation issue, but this is not the behavior I expected from the [description](https://typst.app/docs/reference/layout/place/) of `place`:
```typst
#show parbreak: it => {
  it
  text(red, "BREAK")
}

here is some text #box[aaaaaaaaaa#place(dx:50mm)[here is more text]bbbbbbbbbbbbb]
```
![image](https://user-images.githubusercontent.com/128550314/228363090-3c5b21da-4406-42fd-af20-653ab7f26126.png)
As you can see it is a line break and not a paragraph break because there is no red "BREAK" preceeding `bbbbbbbbbbbbb`.

My expectation would have been that `place` makes no changes to layout regardless of where it's called.

---

**Edit:** `move` does something similar but inserts *two* line breaks:
```typst
#show parbreak: it => {
  it
  text(red, "BREAK")
}

here is some text #box[aaaaaaaaaa#move(dx:50mm)[here is more text]bbbbbbbbbbbbb]
```
![image](https://user-images.githubusercontent.com/128550314/228384558-7f14fcb1-adb3-4986-8723-36292776287e.png)

**Comment by laurmaedje** [comment]
Both are block level elements. To prevent this, you can put the box directly around them.

**Comment by ndtodoroff** [comment]
@laurmaedje That doesn't really work. To be more concrete, what I want is something like a `marginpar` thats aligned with the location in the main text where its declared. I've gotten this to work for a *single* page by using `stack[#a][#b]` with `#a` the maing text and then aligning the margin paragraphs appropriately in `#b`, but this fails when `#a` needs to take up multiple pages. It seems the most straighforward way then would be to just shove text into the margin from directly where its declared using `place`, but that doesn't really work using `box(place(dx: _)[#stuff])` for a couple of reasons (the first two of which are really just syntactic annoyances, but the third is substantial):

 1. Inserting a box separates spaces so for instance `a#box[]b` has no space between "a" and "b", `a #box[]b` and `a#box[] b` has one space between them, but `a #box[] b` has *two* spaces between them. This forces me to write `a#marginpar[#stuff] b` or `a #marginpar[#stuff]b` which isn't great for legibility.
 2. Comments within a line are ignored but a comment spanning multiple lines get treated as a space. So `a/**/b` gives "ab" but
    ```typst
    a//
    b
    ```
    as well as
    ```typst
    a/*
    */b
    ```
    both yields "a  b". This means that
    ```typst
    a//
    #marginpar[#stuff]//
    b
    ```
    is going to yields a double space between "a" and "b".
 3. `place`d content appears *below* where `box` is used:
    ```typst
    There is a box here #box(place(dx: 50mm)[placed content]) now we are after the box \
    Here is the next line to compare with the placed stuff
    ```
    ![image](https://user-images.githubusercontent.com/128550314/229193034-7cd0b953-c4b1-4339-a868-2a35ca9a9c3d.png)
    This spacing seems to be related to the baseline skip, and as far as I know there isn't any way to access that currently so this can't be manually fixed. I can sort of bruteforce it:
    ```typst
    There is a box here #box(baseline: -7.6pt, place(dx: 43mm)[placed content]) now we are after the box \
    Here is the next line to compare with the placed stuff
    ```
    ![image](https://user-images.githubusercontent.com/128550314/229193934-99286153-00fc-469b-aa97-de5730a58498.png)
    but that's not a great solution.

**Comment by laurmaedje** [comment]
Yes, marginal notes aren't really supported by how `place` works, at least at the moment, unfortunately.

**Comment by ntjess** [comment]
FWIW I made a package that provides utilities for margin notes [here](https://github.com/ntjess/typst-drafting) (`@preview/drafting:0.1.1`). You can `locate` where the call to `place` is, and use `layout` to find how far to move to the edge of the page.
